src/services/createRouter.ts

```diff
@@ -168,7 +168,25 @@ export function createRouter<const T extends Routes>(routesOrArrayOfRoutes: T |
     return setRejection(type)
   }
 
-  const find = createRouterFind(routes)
+  const find = <TSource extends RoutesKey<T>>(
+    source: Url | TSource,
+    params: Record<PropertyKey, unknown> = {},
+  ): ResolvedRoute | undefined => {
+    if (!isUrl(source)) {
+      const url = resolve(source, params as any)
+
+      return getResolvedRouteForUrl(routes, url)
+    }
+
+    if (isExternal(source)) {
+      return undefined
+    }
+
+    const url = resolve(source)
+
+    return getResolvedRouteForUrl(routes, url)
```

Comment:
I'm assuming this is how it was already implemented. But isn't `source` already a string at this point? Because the previous if statements.

Reply:
yeah... you said `string`, but it's actually `Url`. Not sure if sending that through `resolve` does something or not 🤔

Reply:
Seems like we could skip the resolve step here and pass it right into getResolvedRouteForUrl but 🤷‍♂️

Reply:
removed